OnProfileUpdate/handler.ts

```diff
@@ -0,0 +1,162 @@
+import * as t from "io-ts";
+import { pipe, flow } from "fp-ts/lib/function";
+import * as O from "fp-ts/lib/Option";
+import * as A from "fp-ts/ReadonlyArray";
+import * as E from "fp-ts/lib/Either";
+import * as TE from "fp-ts/lib/TaskEither";
+import * as RTE from "fp-ts/lib/ReaderTaskEither";
+import { NonNegativeInteger } from "@pagopa/ts-commons/lib/numbers";
+import { DataTableProfileEmailsRepository } from "@pagopa/io-functions-commons/dist/src/utils/unique_email_enforcement/storage";
+import { ProfileEmail } from "@pagopa/io-functions-commons/dist/src/utils/unique_email_enforcement";
+import {
+  ProfileModel,
+  Profile
+} from "@pagopa/io-functions-commons/dist/src/models/profile";
+import { generateVersionedModelId } from "@pagopa/io-functions-commons/dist/src/utils/cosmosdb_model_versioned";
+import { CosmosErrors } from "@pagopa/io-functions-commons/dist/src/utils/cosmosdb_model";
+import { Logger } from "@azure/functions";
+import { FiscalCode } from "../generated/backend/FiscalCode";
+
+const ProfileDocument = t.intersection([
+  ProfileEmail,
+  t.type({
+    isEmailValidated: t.boolean,
+    version: NonNegativeInteger
+  })
+]);
+
+type ProfileDocument = t.TypeOf<typeof ProfileDocument>;
+
+interface IDependencies {
+  readonly dataTableProfileEmailsRepository: DataTableProfileEmailsRepository;
```

Comment:
```suggestion
  readonly dataTableProfileEmailsRepository: IProfileEmailWriter;
```
`DataTableProfileEmailsRepository` is a concrete implementation